VogtObserver2/StockGrabber.cs

```diff
@@ -0,0 +1,66 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace VogtObserver2
+{
+    class StockGrabber : IObservable
+    {
+        private HistoricalDataInfo _hdi = new HistoricalDataInfo();
+        private List<HistoricalDataResponse> _hdr;
+
+        private List<IObserver> _observers = new List<IObserver>();
+
+        public void AddObserver(IObserver o)
+        {
+            _observers.Add(o);
+        }
+
+        public void Notify()
+        {
+            foreach (IObserver observer in _observers)
+            {
+                observer.Update(GetPrice(), GetStockName()); 
+            }
+        }
+
+        public void RemoveObserver(IObserver o)
+        {
+            _observers.Remove(o);
+        }
+
+        public double GetPrice()
+        {
+            
+            var temp = 0.0;
+
+            _hdr = _hdi.HistoricalData($"https://sandbox.iexapis.com/stable/stock/{GetStockName()}/quote?token=Tpk_81485eef3d7041e6bd05ba956b85fa4e");
+
+            foreach (var historicaldata in _hdr)
+            {
+                temp = historicaldata.latestPrice;
+            }
+
+
+            return temp;
+        }
+
+        public string GetStockName() {
```

Comment:
I think a more accurate name is `GetRandomStockName`.

Feels like this should live in Program.cs. Getting a random stock name is more the behavior oof your test program, not a StockGrabber/StockFetcher type of class.